### PR TITLE
Basic support for resizing overlay surfaces in hybrid composition

### PR DESCRIPTION
**Pull request by jason-simmons**
Issue: https://github.com/flutter/flutter/issues/58291
Helps https://github.com/flutter/flutter/issues/59904

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 1]
```diff
@@ -35,7 +38,7 @@
 @SuppressLint("ViewConstructor")
 @TargetApi(19)
 public class FlutterImageView extends View implements RenderSurface {
-  private final ImageReader imageReader;
+  private ImageReader imageReader;
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:41]: nit: `@NonNull`

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 2]
```diff
@@ -63,13 +66,37 @@
...
-  public FlutterImageView(
-      @NonNull Context context, @NonNull ImageReader imageReader, SurfaceKind kind) {
+  public FlutterImageView(@NonNull Context context, int width, int height, SurfaceKind kind) {
+    super(context, null);
+    this.imageReader = createImageReader(width, height);
+    this.kind = kind;
+  }
+
+  @VisibleForTesting
+  FlutterImageView(@NonNull Context context, ImageReader imageReader, SurfaceKind kind) {
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:76]: nit: `@NonNull ImageReader`

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 3]
```diff
@@ -63,13 +66,37 @@
...
     super(context, null);
     this.imageReader = imageReader;
     this.kind = kind;
   }
 
+  @TargetApi(19)
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:82]: nit: `@ NonNull`

File: shell/platform/android/io/flutter/embedding/android/FlutterImageView.java [hunk 4]
```diff
@@ -63,13 +66,37 @@
...
+          height,
+          PixelFormat.RGBA_8888,
+          3,
+          HardwareBuffer.USAGE_GPU_SAMPLED_IMAGE | HardwareBuffer.USAGE_GPU_COLOR_OUTPUT);
+    } else {
+      return ImageReader.newInstance(width, height, PixelFormat.RGBA_8888, 3);
+    }
+  }
+
+  public Surface getSurface() {
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterImageView.java:96]: ditto

File: shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java [hunk 5]
```diff
@@ -855,6 +855,17 @@ public FlutterOverlaySurface createOverlaySurface() {
...
     return platformViewsController.createOverlaySurface();
   }
+
+  @SuppressWarnings("unused")
+  @UiThread
+  public void destroyOverlaySurfaces() {
+    ensureRunningOnMainThread();
+    if (platformViewsController == null) {
+      throw new RuntimeException(
+          "platformViewsController must be set before attempting to position an overlay surface");
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java:865]: nit: `to destroy`

File: shell/platform/android/external_view_embedder/surface_pool.cc [hunk 6]
```diff
@@ -70,6 +70,13 @@ void SurfacePool::RecycleLayers() {
   available_layer_index_ = 0;
 }
 
+void SurfacePool::DestroyLayers(
```
Comment (blasten) [inline shell/platform/android/external_view_embedder/surface_pool.cc:73]: This method could be unit tested

File: shell/platform/android/external_view_embedder/external_view_embedder.cc [hunk 7]
```diff
@@ -238,6 +238,9 @@ void AndroidExternalViewEmbedder::BeginFrame(
     double device_pixel_ratio,
     fml::RefPtr<fml::RasterThreadMerger> raster_thread_merger) {
   Reset();
+  if (frame_size_ != frame_size) {
+    surface_pool_->DestroyLayers(jni_facade_);
```
Comment (blasten) [inline shell/platform/android/external_view_embedder/external_view_embedder.cc:242]: This could be unit tested

**Comment by jason-simmons** [comment]
Done - PTAL

**Pull request merged**